lock-api/src/main/java/com/palantir/lock/client/LockRefreshingLockService.java

```diff
@@ -135,11 +138,25 @@ private void refreshLocks() {
         for (List<LockRefreshToken> tokenBatch : Lists.partition(refreshCopy, REFRESH_BATCH_SIZE)) {
             refreshedTokens.addAll(delegate.refreshLockRefreshTokens(tokenBatch));
         }
+        // Avoid the overhead of constructing a set if we don't need it. Failures should only happen if lock server
+        // fails over (e.g. a new leader was elected), or locks are not being refreshed frequently enough.
+        Set<LockRefreshToken> failedTokens = null;
         for (LockRefreshToken token : refreshCopy) {
             if (!refreshedTokens.contains(token)
                     && toRefresh.contains(token)) {
                 log.warn("failed to refresh lock: {}", token);
                 toRefresh.remove(token);
+                if (failedTokens == null) {
+                    failedTokens = new HashSet<>();
+                }
+                failedTokens.add(token);
+            }
+        }
+        if (failedTokens != null && !failedTokens.isEmpty()) {
+            Set<LockRefreshToken> unmodifiableFailedTokens = Collections.unmodifiableSet(failedTokens);
+            // submit callbacks to the executor so a slow callback doesn't prevent future refreshes
+            for (FailedRefreshCallback callback : failedRefreshCallbacks) {
+                exec.submit(() -> callback.onFailedRefresh(unmodifiableFailedTokens));
```

Comment:
This must use execute, otherwise exceptions will not be recorded.
```suggestion
                exec.execute(() -> callback.onFailedRefresh(unmodifiableFailedTokens));
```

It's unfortunate we don't run baseline errorprone pre-merge, it would have caught this:
https://github.com/palantir/gradle-baseline/blob/7039925b163c647a4e738444a400e67a9fcbec8e/baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ExecutorSubmitRunnableFutureIgnored.java#L33-L42

Reply:
Done